doc/experiment.rst

```diff
@@ -1,326 +1,350 @@
-
-  cd /scratch/smartsim/Training-Run ; /usr/bin/srun --output /scratch/smartsim/Training-Run/Training-Ensemble_3.out --error /scratch/smartsim/Training-Ensemble_3.err --job-name Training-Ensemble_3-CHTN0UI2TRE7 --nodes=1 --ntasks=24 /scratch/pyenvs/smartsim/bin/python ./train-model.py --LR=0.35000000000000003 &
-
-  wait
-
-Prefixing Keys in the Orchestrator
-----------------------------------
-
-If each of multiple ensemble members attempt to use the same code to access their respective models
-in the Orchestrator, the keys by which they do this will overlap and they can end up accessing each
-others' data inadvertently. To prevent this situation, the SmartSim Entity object supports key
-prefixing, which automatically prepends the name of the model to the keys by which it is accessed.
-With this enabled, key overlapping is no longer an issue and ensemble members can use the same code.
-
-Under the hood, calling ensemble.enable_key_prefixing() causes the SSKEYOUT environment variable to
-be set, which in turn causes all keys generated by an ensemble member to be prefixed with its model
-name. Similarly, if the model for the ensemble member has incoming entities (such as those set via
-model.register_incoming_entity() or ensemble.register_incoming_entity()), the SSKEYIN environment
-variable will be set and the keys associated with those inputs will be automatically prefixed. Note
-that entities must register themselves as this is not done by default.
-
-Finally, please note that while prefixing is enabled by default for tensors, datasets, and aggregated
-lists of datasets, a SmartRedis client must manually call Client.use_model_ensemble_prefix() to
-ensure that prefixes are used with models and scripts.
-
-We modify the example above to enable key prefixing as follows:
-
-.. code-block:: bash
-
-  import numpy as np
-  from smartsim import Experiment
-
-  exp = Experiment("Training-Run", launcher="slurm")
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
-
-  # define resources for all ensemble members
-  sbatch = exp.create_batch_settings(nodes=4,
-                                    time="01:00:00",
-                                    account="12345-Cray",
-                                    queue="gpu")
+  .. literalinclude:: ../tutorials/doc_examples/experiment_doc_examples/exp.py
+    :language: python
+    :linenos:
+    :lines: 20-21
 
-  # define how each member should run
-  srun = exp.create_run_settings(exe="python",
-                                exe_args="./train-model.py")
-  srun.set_nodes(1)
-  srun.set_tasks(24)
+  We use the ``Experiment.generate()`` function to create an
```

Comment:
This seems out of place.  Should it be in the generation section?

 We use the ``Experiment.generate()`` function to create an
  output directory for the ``Orchestrator`` `.err` and `.out` log files.

->

Because we used the ``Experiment.generate()`` function to create an
  output directory for the ``Experiment`` entities, `.err` and `.out` log files will be located in entity subdirectories within the ``Experiment`` directory.

Reply:
yes I will move it